pymongo/asynchronous/client_bulk.py

```diff
@@ -133,7 +133,10 @@ def add_insert(self, namespace: str, document: _DocumentOut) -> None:
         validate_is_document_type("document", document)
         # Generate ObjectId client side.
         if not (isinstance(document, RawBSONDocument) or "_id" in document):
-            document["_id"] = ObjectId()
+            new_document = {"_id": ObjectId()}
```

Comment:
This implementation is also incomplete because it does not put the `_id` field first if the user supplies it. For example when inserting `{"a": 1, "_id": 2}`. We should add tests for this case for insert/bulk/clientBulk as well.

Reply:
We've decided to make re-ordering user-supplied `_id` fields optional due to the complexity of doing so across different driver implementations. We can do it in PyMongo if we want, but it won't be standard across all drivers.